src/nexgen/nxs_write/NXclassWriters.py

```diff
@@ -59,48 +58,48 @@ def write_NXdata(
     goniometer: Dict,
     data_type: str,
     coord_frame: str,
-    scan_range: Union[Tuple, np.ndarray],
-    scan_axis: str = None,
+    scan_range: Dict[str, Union[Tuple, np.ndarray]],
 ):
     """
     Write NXdata group at entry/data
 
     Args:
-        nxsfile (h5py.File):        NeXus file to be written.
-        datafiles (List):           List of Path objects.
-        goniometer (Dict):          Dictionary containing all the axes information.
-        data_type (str):            Images or events.
-        coord_frame (str):          Coordinate system the axes are currently in.
-        scan_range (Tuple|array):   If writing events, this is just a (start, end) tuple.
-        scan_axis (str):            Rotation axis.
+        nxsfile (h5py.File):        NeXus file to be written
+        datafiles (List):           List of Path objects
+        goniometer (Dict):          Dictionary containing all the axes information
+        data_type (str):            Images or events
+        coord_frame (str):          Coordinate system the axes are currently in
+        scan_range (Dict):          A dictionary of the scan axes and their movements
     """
     NXclass_logger.info("Start writing NXdata.")
     # Check that a valid datafile_list has been passed.
     assert len(datafiles) > 0, "Please pass at least a list of one HDF5 data file."
 
-    # If scan_axis hasn't been passed, identify it.
-    if not scan_axis:
-        scan_axis = find_scan_axis(
-            goniometer["axes"],
-            goniometer["starts"],
-            goniometer["ends"],
-            goniometer["types"],
-        )
-
     # Create NXdata group, unless it already exists, in which case just open it.
     nxdata = nxsfile.require_group("/entry/data")
     create_attributes(
         nxdata,
-        ("NX_class", "axes", "signal", scan_axis + "_indices"),
+        ("NX_class", "signal"),
         (
             "NXdata",
-            scan_axis,
             "data",
-            [
-                0,
-            ],
         ),
     )
+    if len(scan_range.keys()) == 1:
```

Comment:
So I think for what I am currently passing in from `artemis` if I left it at what this was the scan axis will be set to one of "sam_x" or "sam_y". We actually want it set to `omega`, right? I'm not sure if the logic for adding the omega in should be done in `artemis` or `nexgen`. This will actually be solved in when we move to the 3D gridscan as Artemis will then need to provide `omega`

Reply:
I agree it should be in `nexgen`. Maybe we should have a chat about what the best interface looks like for setting scan info then as we need top provide it in such a way that `nexgen` knows it's a `gridscan` and to create the `chi`/`phi` part itself.